packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -261,6 +266,22 @@ export const Popover = ({
   // re-attached on every render.
   const popperRef = useLatest(popperElement);
 
+  useEffect(() => {
+    if (!isOpen) {
+      return undefined;
+    }
+    const handleEscapePress = (event: Event) => {
+      if (isEscape(event)) {
+        setOpen(false);
+      }
+    };
+
+    document.addEventListener('keydown', handleEscapePress);
+    return () => {
+      document.removeEventListener('keydown', handleEscapePress);
+    };
+  }, [isOpen]);
```

Comment:
Could you add a test case for the close-on-ESC, please?

Reply:
Will do.